core-api/src/main/java/com/optimizely/ab/optimizelydecision/OptimizelyDecision.java

```diff
@@ -0,0 +1,148 @@
+/**
+ *
+ *    Copyright 2020, Optimizely and contributors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package com.optimizely.ab.optimizelydecision;
+
+import com.optimizely.ab.OptimizelyUserContext;
+import com.optimizely.ab.optimizelyjson.OptimizelyJSON;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public class OptimizelyDecision {
+    @Nullable
+    private final String variationKey;
+
+    private final boolean enabled;
+
+    @Nonnull
+    private final OptimizelyJSON variables;
+
+    @Nullable
+    private final String ruleKey;
+
+    @Nonnull
+    private final String flagKey;
+
+    @Nonnull
+    private final OptimizelyUserContext userContext;
+
+    @Nonnull
+    private List<String> reasons;
+
+
+    public OptimizelyDecision(@Nullable String variationKey,
```

Comment:
I will prefer the following sequence for args.  Just a suggestion.
1. userContext
2. flagKey
3. rule_key
4. variation_key
5. enabled (It may come after flagKey as well.)
6. reasons

Reply:
It makes sense in general. Here, I intended with the current order for more important decision info first and then debug/info values following.